tensorflow/core/kernels/mkl_concat_op.cc

```diff
@@ -342,35 +361,61 @@ class MklConcatFwdPrimitive : public MklPrimitive {
       std::shared_ptr<mkldnn::memory::desc> source_md(
           new memory::desc(srcs_md[i].data));
       context_.src_md.push_back(source_md);
-
+#ifdef ENABLE_MKLDNN_V1
+      std::shared_ptr<mkldnn::memory> src_mem(
+          new mkldnn::memory(*source_md, cpu_engine_, DummyData));
+#else
       std::shared_ptr<mkldnn::memory::primitive_desc> src_mpd(
           new memory::primitive_desc(*source_md, cpu_engine_));
       context_.src_pd_shdptr.push_back(src_mpd);
 
       std::shared_ptr<mkldnn::memory> src_mem(
           new mkldnn::memory(*src_mpd, DummyData));
+      context_.src_pd.push_back(*context_.src_pd_shdptr[i]);
+#endif  // ENABLE_MKLDNN_V1
       context_.data_mem_shdptr.push_back(src_mem);
-
       context_.data_mem.push_back(*context_.data_mem_shdptr[i]);
-      context_.src_pd.push_back(*context_.src_pd_shdptr[i]);
     }
-    // Create a concat primitive descriptor
+// Create a concat primitive descriptor
+#ifdef ENABLE_MKLDNN_V1
+    context_.fwd_pd.reset(new concat::primitive_desc(
+        concat_fwd_dims.concat_dims, context_.src_md, cpu_engine_));
+#else
     context_.fwd_pd.reset(new concat::primitive_desc(
         concat_fwd_dims.concat_dims, context_.src_pd));
+#endif  // ENABLE_MKLDNN_V1
 
     // Store the expected memory format
     context_.dst_md.reset(new memory::desc({concat_fwd_dims.dst_dims},
                                            MklDnnType<T>(),
                                            concat_fwd_dims.mkl_common_format));
+#ifdef ENABLE_MKLDNN_V1
+    // Create memory primitive based on dummy data
+    context_.dst_mem.reset(
+        new memory(*context_.dst_md, cpu_engine_, DummyData));
+#else
     context_.dst_pd.reset(
         new memory::primitive_desc(*context_.dst_md, cpu_engine_));
 
     // Create memory primitive based on dummy data
     context_.dst_mem.reset(new memory(*context_.dst_pd, DummyData));
+#endif  // ENABLE_MKLDNN_V1
```

Comment:
I think we can do 
```c++
#ifndef ENABLE_MKLDNN_V1
    context_.dst_pd.reset(
        new memory::primitive_desc(*context_.dst_md, cpu_engine_));	        new memory::primitive_desc(*context_.dst_md, cpu_engine_));
#endif  // !ENABLE_MKLDNN_V1
    // Create memory primitive based on dummy data
    context_.dst_mem.reset(
        MEMORY_CONSTRUCTOR(*context_.dst_md, cpu_engine_, DummyData));
```

Reply:
i will check and do

Reply:
The memory primitive descriptor is required to create the memory primitive in !ENABLE_MKLDNN_V1 case. Here we are caching both the memory primitive and primitive descriptor in different class variables so the above suggestion is not possible.

Reply:
Thank you for the explanation!